### PR TITLE
TVAULT-5881 TVM uses outdated TLS protocols when HTTPS is activated f…

### PR DESCRIPTION
**Pull request by pushkar-mahajan**
…or the Trilio API

@abhijeetpatra @rajneeshkapoor could you please review this

### CODE REVIEW

File: TVOAppliance/hf_upgrade.sh [hunk 1]
```diff
@@ -95,6 +95,31 @@ function reconfigure_s3_service_path()
...
+
+#function to keep only tls1_2 access
+function limit_access_to_tls1_2()
+{
+	config_file="/etc/nginx/nginx.conf"
+	config_line="ssl_protocols TLSv1.2;"
+
+	# Check if the line already exists in the file
+	if grep -q "$config_line" "$config_file"; then
+		echo "Line already exists in $config_file"
```
Comment (abhijeetpatra) [inline TVOAppliance/hf_upgrade.sh]: this output msg won't make any sense to the user.
Mention below msg instead:
if already exists, "SSL protocol set to TLSv1.2 already"
else, "Setting the SSL protocol to TLSv1.2"

File: TVOAppliance/hf_upgrade.sh [hunk 2]
```diff
@@ -95,6 +95,31 @@ function reconfigure_s3_service_path()
...
+	else
+	  # Add the line to the http section
+	  	sed -i '/^\s*http {/a \    ssl_protocols TLSv1.2;' "$config_file"
+	  	echo "Added line to $config_file"
```
Comment (abhijeetpatra) [inline TVOAppliance/hf_upgrade.sh]: check my comment above

File: TVOAppliance/hf_upgrade.sh [hunk 3]
```diff
@@ -95,6 +95,31 @@ function reconfigure_s3_service_path()
...
+	  	pcs resource restart lb_nginx-clone
```
Comment (abhijeetpatra) [inline TVOAppliance/hf_upgrade.sh]: don't restart it here, move it to the section below where we check if the appliance is already configured.
restart the pcs resource in that section and mention the resource name in the output msg.

File: TVOAppliance/hf_upgrade.sh (hunk 1 above)
Comment (pushkar-mahajan) [reply TVOAppliance/hf_upgrade.sh]: Done

File: TVOAppliance/hf_upgrade.sh (hunk 2 above)
Comment (pushkar-mahajan) [reply TVOAppliance/hf_upgrade.sh]: Done

File: TVOAppliance/hf_upgrade.sh (hunk 3 above)
Comment (pushkar-mahajan) [reply TVOAppliance/hf_upgrade.sh]: moved

**Review by abhijeetpatra — APPROVED** [review]

File: TVOAppliance/hf_upgrade.sh [hunk 4]
```diff
@@ -95,6 +95,25 @@ function reconfigure_s3_service_path()
...
 }
+
+#function to keep only tls1_2 access
+function limit_access_to_tls1_2()
+{
+	config_file="/etc/nginx/nginx.conf"
+	config_line="ssl_protocols TLSv1.2;"
+
+	# Check if the line already exists in the file
+	if grep -q "$config_line" "$config_file"; then
```
Comment (pushkar-mahajan) [inline TVOAppliance/hf_upgrade.sh]: add http check in the grep command 
grep -A4 http nginx.conf | grep -q "ssl_protocols TLSv1.2" 
add regex for HTTP<space> {

File: TVOAppliance/hf_upgrade.sh [hunk 5]
```diff
@@ -95,6 +95,25 @@ function reconfigure_s3_service_path()
...
+		echo "SSL protocol set to TLSv1.2 already."
+	else
+	  # Add the line to the http section
+	  	sed -i '/^\s*http {/a \    ssl_protocols TLSv1.2;' "$config_file"
```
Comment (pushkar-mahajan) [inline TVOAppliance/hf_upgrade.sh]: regex for HTTP { where space may or may not exist

**Review by rajneeshkapoor — APPROVED** [review]

**Pull request merged**